files/default/setup-ephemeral-drives.sh

```diff
@@ -13,124 +11,189 @@ set -x
+  fi
+}
+
+function activate_lvm {
+  LVM_STATE=$(lvs "${LVM_PATH}" --nosuffix --noheadings -o lv_attr | xargs | cut -c5)
+  log "Found LVM (${LVM_PATH}) in state (${LVM_STATE})"
+
+  if [[ "${LVM_STATE}" != "${LVM_ACTIVE_STATE}" ]]; then
+    log "Activating LVM (${LVM_PATH})"
+    LVM_ACTIVATE_COMMAND="lvchange -ay ${LVM_PATH}"
+    if ! ${LVM_ACTIVATE_COMMAND}; then
+      error_exit "Failed to activate LVM"
+    else
+      log "LVM (${LVM_PATH}) activated successfully"
     fi
+  fi
+}
+
+function format_lvm {
+  LVM_FS_TYPE=$(lsblk "${LVM_PATH}" --noheadings -o FSTYPE | xargs)
+  log "Found LVM (${LVM_PATH}) FS type (${LVM_FS_TYPE})"
 
-    # sleep 10 seconds to let partitions settle (bug?)
-    sleep 10
-
-    # Setup LVM
-    RC=0
-    pvcreate -y $PARTITIONS || RC=1
-    vgcreate vg.01 $PARTITIONS || RC=1
-    lvcreate -i $NUM_DEVS -I 64 -l 100%FREE -n lv_ephemeral vg.01 || RC=1
-    if [ "${encrypted_ephemeral}" == "true" ]; then
-      modprobe brd || RC=1
-      mkfs -q /dev/ram1 1024 || RC=1
-      mkdir -p /root/keystore || RC=1
-      mount /dev/ram1 /root/keystore || RC=1
-      dd if=/dev/urandom of=/root/keystore/keyfile bs=1024 count=4 || RC=1
-      chmod 0400 /root/keystore/keyfile || RC=1
-      cryptsetup -q luksFormat /dev/vg.01/lv_ephemeral /root/keystore/keyfile || RC=1
-      cryptsetup -d /root/keystore/keyfile luksOpen /dev/vg.01/lv_ephemeral ephemeral_luks || RC=1
-      mkfs.ext4 /dev/mapper/ephemeral_luks || RC=1
-      mount -v -t ext4 -o noatime,nodiratime /dev/mapper/ephemeral_luks ${ephemeral_dir} || RC=1
+  if [[ "${LVM_FS_TYPE}" != "${FS_TYPE}" ]]; then
+    log "Formatting LVM (${LVM_PATH}) with FS type (${FS_TYPE})"
+    LVM_FORMAT_COMMAND="mkfs -t ${FS_TYPE} ${LVM_PATH}"
+    if ! ${LVM_FORMAT_COMMAND}; then
+      error_exit "Failed to format LVM"
     else
-      mkfs.ext4 /dev/vg.01/lv_ephemeral || RC=1
-      echo "/dev/vg.01/lv_ephemeral ${ephemeral_dir} ext4 noatime,nodiratime 0 0" >> /etc/fstab || RC=1
-      mount -v ${ephemeral_dir} || RC=1
+      log "LVM (${LVM_PATH}) formatted successfully"
     fi
+    sync
+    sleep 1
+  else
+    log "LVM (${LVM_PATH}) already formatted with FS type (${LVM_FS_TYPE})"
   fi
-  chmod 1777 ${ephemeral_dir} || RC=1
-  if [ $RC -ne 0 ]; then
-    error_exit "Failed to create LVM stripe and/or format ephemeral volume."
+}
+
+function create_mount {
+  mkdir -p "${INPUT_MOUNTPOINT}"
+}
+
+function set_mount_permission {
+  chmod 1777 "${INPUT_MOUNTPOINT}"
+}
+
+function mount_lvm {
+  LVM_MOUNTPOINT=$(lsblk "${LVM_PATH}" -o MOUNTPOINT --noheadings | xargs)
+
+  if [[ -z ${LVM_MOUNTPOINT} ]]; then
+    log "LVM (${LVM_PATH}) not mounted, mounting on (${INPUT_MOUNTPOINT})"
+    create_mount
+    LVM_MOUNT_COMMAND="mount -v -t ${FS_TYPE} -o ${MOUNT_OPTIONS} ${LVM_PATH} ${INPUT_MOUNTPOINT}"
+    if ! ${LVM_MOUNT_COMMAND}; then
+      error_exit "Failed to mount LVM"
+    else
+      log "LVM (${LVM_PATH}) mounted successfully"
+    fi
+    set_mount_permission
```

Comment:
we might avoid create_mount and set_mount_permission function, given that are 1line functions and used in a single place.

Reply:
done